api/apps/api/src/modules/projects/job-status/job-status.view.api.entity.ts

```diff
@@ -0,0 +1,76 @@
+import { ViewColumn, ViewEntity } from 'typeorm';
+import { JobType } from '@marxan-api/modules/projects/job-status/jobs.enum';
+import { JobStatus } from '@marxan-api/modules/scenarios/scenario.api.entity';
+import { API_EVENT_KINDS } from '@marxan/api-events';
+
+@ViewEntity({
+  expression: `
+    SELECT
+      DISTINCT ON ("jobType", topic) "jobType",
+      api_events.topic AS "scenarioId",
+      projects.id AS "projectId",
+      api_events.kind
+    FROM
+      api_events
+      INNER JOIN scenarios ON api_events.topic = scenarios.id
+      INNER JOIN projects ON projects.id = scenarios.project_id
+      CROSS JOIN LATERAL SUBSTRING(
+        api_events.kind
+        FROM
+          'scenario.#"[^.]*#"%' FOR '#'
+      ) AS "jobType"
+    ORDER BY
+      "jobType",
+      api_events.topic,
+      api_events.timestamp DESC;
+  `,
+})
+export class ScenarioJobStatus {
+  @ViewColumn()
+  jobType!: JobType;
+
+  @ViewColumn()
+  kind!: API_EVENT_KINDS;
+
+  get jobStatus(): JobStatus | undefined {
+    // I didn't use CASE ... THEN in the SQL as I wanted to enforce the compiler check on the mapping
+    return eventToJobStatusMapping[this.kind];
+  }
+
+  @ViewColumn()
+  scenarioId!: string;
+
+  @ViewColumn()
+  projectId!: string;
+}
+
+const eventToJobStatusMapping: Record<
+  API_EVENT_KINDS,
+  JobStatus | undefined
+> = {
+  [API_EVENT_KINDS.project__protectedAreas__failed__v1__alpha]: undefined,
+  [API_EVENT_KINDS.project__protectedAreas__finished__v1__alpha]: undefined,
+  [API_EVENT_KINDS.project__protectedAreas__submitted__v1__alpha]: undefined,
+  [API_EVENT_KINDS.scenario__costSurface__costUpdateFailed__v1_alpha1]:
+    JobStatus.failure,
+  [API_EVENT_KINDS.scenario__costSurface__finished__v1_alpha1]: JobStatus.done,
+  [API_EVENT_KINDS.scenario__costSurface__shapeConversionFailed__v1_alpha1]:
+    JobStatus.failure,
+  [API_EVENT_KINDS.scenario__costSurface__shapeConverted__v1_alpha1]:
+    JobStatus.running,
```

Comment:
Seems to cover existing https://github.com/Vizzuality/marxan-cloud/pull/253/files#diff-d0dcb799abb6600133ff14d39c366ce3ccad3324da0c01e42732a9745f4f1682

One of those should be enough (the linked one tho seems to be more readable ;) )

Reply:
It contains much less events (e.g. missing `shapeConversionFailed`) and does not have the compiler enforcement that I wanted.